test/controllers/sessions_controller_test.rb

```diff
@@ -88,6 +88,30 @@ class SessionsControllerTest < Devise::ControllerTestCase
     assert_equal 204, @response.status
   end
 
+  test "#destroy returns 204 status if the requested format is not navigational" do
```

Comment:
This behavior is already tested in the test above. IMHO changing the test name would be enough.

`#destroy doesn't set the flash and returns 204 status if the requested format is not navigational`